test/Microsoft.VisualStudio.Editor.Razor.Test/DefaultRazorSyntaxFactsServiceTest.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.Razor.Language;
+using Xunit;
+
+namespace Microsoft.VisualStudio.Editor.Razor
+{
+    public class DefaultRazorSyntaxFactsServiceTest
+    {
+        [Fact]
+        public void GetClassifiedSpans_ReturnsExpectedSpans()
+        {
+            // Arrange
+            var expectedSpans = new[]
+            {
+                new ClassifiedSpan(new SourceSpan("test.cshtml", 0, 0, 0, 5), new SourceSpan("test.cshtml", 0, 0, 0, 5), SpanKind.Markup, BlockKind.Tag, AcceptedCharacters.Any),
+                new ClassifiedSpan(new SourceSpan("test.cshtml", 5, 0, 5, 6), new SourceSpan("test.cshtml", 0, 0, 0, 42), SpanKind.Markup, BlockKind.Markup, AcceptedCharacters.Any),
+                new ClassifiedSpan(new SourceSpan("test.cshtml", 34, 1, 27, 2), new SourceSpan("test.cshtml", 0, 0, 0, 42), SpanKind.Markup, BlockKind.Markup, AcceptedCharacters.Any),
+                new ClassifiedSpan(new SourceSpan("test.cshtml", 36, 2, 0, 6), new SourceSpan("test.cshtml", 36, 2, 0, 6), SpanKind.Markup, BlockKind.Tag, AcceptedCharacters.Any),
+            };
+            var codeDocument = GetCodeDocument(
+@"<div>
+    <taghelper></taghelper>
+</div>");
+            var syntaxTree = codeDocument.GetSyntaxTree();
+            var service = new DefaultRazorSyntaxFactsService();
+
+            // Act
+            var spans = service.GetClassifiedSpans(syntaxTree);
+
+            // Assert
+            Assert.Equal(expectedSpans, spans);
+        }
+
+        [Fact]
+        public void GetTagHelperSpans_ReturnsExpectedSpans()
+        {
+            // Arrange
+            var codeDocument = GetCodeDocument(
+@"<div>
+    <taghelper></taghelper>
+</div>");
+            var tagHelperContext = codeDocument.GetTagHelperContext();
+            var expectedSourceSpan = new SourceSpan("test.cshtml", 11, 1, 4, 23);
+            var syntaxTree = codeDocument.GetSyntaxTree();
+            var service = new DefaultRazorSyntaxFactsService();
+
+            // Act
+            var spans = service.GetTagHelperSpans(syntaxTree);
+
+            // Assert
+            var actualSpan = Assert.Single(spans);
+            Assert.Equal(expectedSourceSpan, actualSpan.Span);
+            Assert.Equal(tagHelperContext.TagHelpers, actualSpan.TagHelpers);
+            Assert.Equal(tagHelperContext.Prefix, actualSpan.Binding.TagHelperPrefix);
```

Comment:
Intentionally not verifying the entire `TagHelperBinding` here because it is constructible from the information being verified here. This seemed cleaner.